### DISCUSSION TITLE
What is the equivalent of EVP_DigestSqueeze for KMAC-XOF?

### DISCUSSION THREAD

**Discussion by bartonjs** [openssl/openssl#25398] [Q&A] [answered]
[Back in 2021](https://github.com/openssl/openssl/issues/13232#issuecomment-779204745) @levitte said:

> In the section on KMAC, [SP 800-185](https://doi.org/10.6028/NIST.SP.800-185) talks about exactly the scenario discussed here, where output may start to be pulled before knowing the desired output length, and how XOF is an applicable method to do so.

SHAKE (an EVP_MD) gained that functionality via EVP_DigestSqueeze, but KMAC-XOF (an EVP_MAC) doesn't seem to have, despite them both being cSHAKE under the covers.

Am I just missing something obvious as to how to use KMAC-XOF in a streaming manner?

**Comment by t8m** [comment, marked as answer]
Unfortunately KMAC does not provide such API yet.

**Comment by paulidale** [comment]
Issue raised to implement this: #25400
